test/integration/singlecluster/controller/jobs/job/job_controller_test.go

```diff
@@ -2667,3 +2668,325 @@ var _ = ginkgo.Describe("Job controller when TopologyAwareScheduling enabled", g
 		})
 	})
 })
+
+var _ = ginkgo.Describe("Job controller with ObjectRetentionPolicies", ginkgo.Ordered, ginkgo.ContinueOnFailure, func() {
```

Comment:
Please run these a bunch of times (let's say ~100) to see if obviously flaky

Reply:
running these tests 100 times did not produce any failures, running e2e now